compiler/ilgen/VirtualMachineOperandStack.hpp

```diff
@@ -0,0 +1,186 @@
+    * @param b builder object to use for any operations used to implement the push (e.g. update the top of stack)
+    * @param value expression to push onto the simulated operand stack
+    */
+   virtual void Push(TR::IlBuilder *b, TR::IlValue *value);
+
+   /**
+    * @brief Pops an expression from the top of the simulated operand stack
+    * @param b builder object to use for any operations used to implement the pop (e.g. to update the top of stack)
+    * @returns expression popped from the stack
+    */
+   virtual TR::IlValue *Pop(TR::IlBuilder *b);
+
+   /**
+    * @brief Returns the expression at the top of the simulated operand stack
+    * @returns the expression at the top of the operand stack
+    */
+   virtual TR::IlValue *Top();
+
+   /**
+    * @brief Returns an expression below the top of the simulated operand stack
+    * @param depth number of values below top (Pick(0) is same as Top())
+    * @returns the requested expression from the operand stack
+    */
+   virtual TR::IlValue *Pick(int32_t depth);
+
+   /**
+    * @brief Removes some number of expressions from the operand stack
+    * @param b builder object to use for any operations used to implement the drop (e.g. to update the top of stack)
+    * @param depth how many values to drop from the stack
+    */
+   virtual void Drop(TR::IlBuilder *b, int32_t depth);
+
+   /**
+    * @brief Duplicates the expression on top of the simulated operand stack
+    * @param b builder object to use for any operations used to duplicate the expression (e.g. to update the top of stack)
+    */
+   virtual void Dup(TR::IlBuilder *b);
+
+   /**
+    * @brief virtual function subclasses can use to configure virtual machine stack growth direction
+    * @returns true if the virtual machine stack grows towards larger addresses, false otherwise
+    */
+   virtual bool growsUp() { return true; }
+
+   /**
+    * @brief virtual function subclasses can use to configure virtual machine stack stack offset 
+    * @returns difference in elements between initial stack pointer and actual bottom of stack
+    * Some stacks Push by incrementing stack pointer then storing, some by storing and then
+    * incrementing stack pointer. In the first case, stackPtrStartingOffset() should return -1
+    * because the stack pointer initially points one element below the bottom of the stack.
+    * In the second case, startPtrStartingOffset() should return 0, because the stack pointer
+    * initially points at the bottom of the stack. Other values are possible but would be
+    * considered highly unusual.
+    * Default assumption is the first case, so return -1.
+    */
+   virtual int32_t stackPtrStartingOffset() { return -1; }
+
+   protected:
+   void copyTo(OMR::VirtualMachineOperandStack *copy);
+   void checkSize();
+   void grow();
+
+   private:
+   TR::MethodBuilder *_mb;
+   OMR::VirtualMachineRegister *_stackTopRegister;
+   int32_t _stackMax;
+   int32_t _stackTop;
```

Comment:
A highly pedantic question (and please be gentle/genteel in your response to this one) 

Should you be accounting for the possibility that an operand stack for some esoteric language might be allowed to have more than 32 bits worth of elements ?